frontend/src/components/Recommendation/RestaurantRecommendations.jsx

```diff
@@ -0,0 +1,122 @@
+import React, { useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+
+import axios from 'axios';
+
+import Container from '@mui/material/Container';
+import Typography from '@mui/material/Typography';
+import Button from '@mui/material/Button';
+import List from '@mui/material/List';
+import Divider from '@mui/material/Divider';
+
+import RestaurantElement from './RestaurantElement';
+
+export const RestaurantRecommendations = () => {
+
+  const { location } = useParams(); // Extract the location parameter from the current route
+
+  // recommendations is an array of objects, each object represents a restaurant
+  const [recommendations, setRecommendations] = useState([]);
+
+  // selectedIndex is the index of the selected restaurant to display details
+  const [selectedIndex, setSelectedIndex] = React.useState(0);
+
+  /**
+   * Handles the click event of a restaurant.
+   * @param {React.MouseEvent<HTMLDivElement, MouseEvent>} event - The click event.
+   * 
+   * @param {number} index - The index of the selected restaurant.
+   * 
+   * @returns {void}
+   */
+  const handleListItemClick = (
+    event, // React.MouseEvent<HTMLDivElement, MouseEvent>
+    index, // number
+  ) => {
+    setSelectedIndex(index);
+  };
+
+  // checked is the index of the selected restaurant for further usage
+  const [checked, setChecked] = React.useState([]);
+
+  /**
+   * Handles the selection of a restaurant.
+   *
+   * @param {number} value - The index of the selected restaurant.
+   */
+  const handleToggleRestaurant = (value) => () => {
+    const currentIndex = checked.indexOf(value);
+    const newChecked = [...checked];
+
+    if (currentIndex === -1) {
+      newChecked.push(value);
+    } else {
+      newChecked.splice(currentIndex, 1);
+    }
+
+    setChecked(newChecked);
+  };
+
+  /**
+   * Capitalizes the first letter of each word in a string.
+   * 
+   * @param {string} str - The string to capitalize.
+   * 
+   * @returns {string} The capitalized string.
+   * 
+   * @example
+   * capitalizeEveryWord('hello world');
+   * // Returns 'Hello World'
+   * 
+   */
+  function capitalizeEveryWord(str) {
+    return str.split(' ')
+      .map(word => word.charAt(0).toUpperCase() + word.slice(1).toLowerCase())
+      .join(' ');
+  }
+
+  useEffect(() => {
+    // Fetch recommendations from the API
+    const fetchRecommendations = async () => {
+      try {
+        const url = 'http://localhost:3000/api/recommendation/restaurant/'+location;
```

Comment:
Should we call this API from the `<APIProvider />`?
Example usage is in `<AuthProvider />`
![image](https://github.com/UOA-CS732-SE750-Students-2024/project-group-careless-crayfish/assets/29388401/2b4d812d-5cd3-4564-96a2-7b3efbd34fef)


`<APIProvider />` does the following:
1. get/post/patch/put/delete rest api handlers which can be used to call our backend api
2. handles error from api response and has logic to display error message on snackbar
![image](https://github.com/UOA-CS732-SE750-Students-2024/project-group-careless-crayfish/assets/29388401/809f210a-b0ca-463e-b6e0-f7b3542a2a68)

UI looks like:
![image](https://github.com/UOA-CS732-SE750-Students-2024/project-group-careless-crayfish/assets/29388401/69e5b0dc-08c3-4376-b88e-d85e958a1ab5)

3. it is unit tested
4. It provides **global states** so you can access your api responses (data and error) from anywhere
![image](https://github.com/UOA-CS732-SE750-Students-2024/project-group-careless-crayfish/assets/29388401/af8ab648-3a3d-4355-84d2-82462f351633)